src/libstore/derived-path.hh

```diff
@@ -25,6 +25,9 @@ struct DerivedPathOpaque {
     nlohmann::json toJSON(ref<Store> store) const;
     std::string to_string(const Store & store) const;
     static DerivedPathOpaque parse(const Store & store, std::string_view);
+
+    bool operator < (const DerivedPathOpaque & b) const
+    { return path < b.path; }
```

Comment:
It might be good to standardize on @thufschmitt's `GENERATE_CMP` (from https://github.com/NixOS/nix/blob/master/src/libutil/comparator.hh) for these?

Reply:
In this case:
```c++
GENERATE_CMP(DerivedPathOpaque, me->path)
```